assets/js/googlesitekit/modules/datastore/sharing-settings.test.js

```diff
@@ -612,5 +616,143 @@ describe( 'core/modules sharing-settings', () => {
+				registry
+					.dispatch( CORE_MODULES )
+					.setSharingManagement( 'analytics', 'all_admins' );
+				expect(
+					registry
+						.select( CORE_MODULES )
+						.haveSharingSettingsExpanded( 'management' )
+				).toBe( true );
+
+				// False after updating module's `management` back to original server value on client.
+				registry
+					.dispatch( CORE_MODULES )
+					.setSharingManagement( 'analytics', 'owner' );
+				expect(
+					registry
+						.select( CORE_MODULES )
+						.haveSharingSettingsExpanded( 'management' )
+				).toBe( false );
+
+				// False after updating module's `management` from `all-admins` to `owner`.
+				registry
+					.dispatch( CORE_MODULES )
+					.setSharingManagement( 'search-console', 'owner' );
+				expect(
+					registry
+						.select( CORE_MODULES )
+						.haveSharingSettingsExpanded( 'management' )
+				).toBe( false );
+			} );
+
+			it( "informs whether the `sharedRoles` settings for any module contain roles that haven't been previously selected", async () => {
+				global[ dashboardSharingDataBaseVar ] = undefined;
+
+				// Initially undefined.
+				expect(
+					registry
+						.select( CORE_MODULES )
+						.haveSharingSettingsExpanded( 'sharedRoles' )
+				).toBeUndefined();
+
+				global[ dashboardSharingDataBaseVar ] = dashboardSharingData;
+				registry.select( CORE_MODULES ).getSharingSettings();
+
+				// Still false after getting the sharing settings from the global variable.
+				expect(
+					registry
+						.select( CORE_MODULES )
+						.haveSharingSettingsExpanded( 'sharedRoles' )
+				).toBe( false );
+
+				// False after removing some of the module's existing `sharedRoles` on the client.
+				registry
+					.dispatch( CORE_MODULES )
+					.setSharedRoles( 'search-console', [ 'editor' ] );
+				expect(
+					registry
+						.select( CORE_MODULES )
+						.haveSharingSettingsExpanded( 'sharedRoles' )
+				).toBe( false );
+
+				// False after adding back the removed existing role to the module's `sharedRoles` on the client.
+				registry
+					.dispatch( CORE_MODULES )
+					.setSharedRoles( 'search-console', [
+						'editor',
+						'subscriber',
+					] );
+				expect(
+					registry
+						.select( CORE_MODULES )
+						.haveSharingSettingsExpanded( 'sharedRoles' )
+				).toBe( false );
+
+				// True after adding new non-existing role to the module's `sharedRoles` on the client.
+				registry
+					.dispatch( CORE_MODULES )
+					.setSharedRoles( 'search-console', [
+						'editor',
+						'administrator',
+					] );
+				expect(
+					registry
+						.select( CORE_MODULES )
+						.haveSharingSettingsExpanded( 'sharedRoles' )
+				).toBe( true );
+
+				// True when adding a new role when the module's `sharedRoles` is empty.
+				registry
+					.dispatch( CORE_MODULES )
+					.setSharedRoles( 'adsense', [ 'editor' ] );
+				expect(
+					registry
+						.select( CORE_MODULES )
+						.haveSharingSettingsExpanded( 'sharedRoles' )
+				).toBe( true );
```

Comment:
Because `haveSharingSettingsExpanded` is already returning `true`, this block doesn't actually go so far as to prove that the `setSharedRoles( 'adsense', [ 'editor' ] )` actually changes the subsequent return value of `haveSharingSettingsExpanded`.

In order to do so, `haveSharingSettingsExpanded` should return `false` first. I'd suggest either adding a block like this immediately above it, or possibly refactoring the test into smaller, more granular tests:

```js
				// False after restoring the module's original `sharedRoles` on the client.
				registry
					.dispatch( CORE_MODULES )
					.setSharedRoles( 'search-console', [
						'editor',
						'subscriber',
					] );
				expect(
					registry
						.select( CORE_MODULES )
						.haveSharingSettingsExpanded( 'sharedRoles' )
				).toBe( false );
```